src/gsi-ncdiag/subset_files.py

```diff
@@ -16,30 +16,46 @@ def subset(infile, nlocsout, suffix, geofile, diagfile):
     ncout = nc.Dataset(outfile, 'w')
     # I think this has to be called before each call to random.sample to make it the same random set
     random.seed(5)
+    # Get number of records (nrecsin)
+    recnum = ncin.variables['record_number@MetaData'][:]
+    recnum_uniq = np.unique(recnum)
+    nrecsin = len(recnum_uniq)
     nlocsin = len(ncin.dimensions['nlocs'])
     nvars = len(ncin.dimensions['nvars'])
-    nsamples = nlocsout
-    npossible = nlocsin
+    if nrecsin > 100.:
+        nsamples = int(-(-nlocsout/10.))  # just picked 10 randomly
```

Comment:
why negative negative?

Reply:
@CoryMartin-NOAA This is your original coding.  You are right,  the double negative can be removed.

Reply:
Ha! Well I don't know what I was thinking here originally then.

Reply:
@CoryMartin-NOAA I probably would do the same thing (thinking about the math, not the coding)  your original design about setting the sample size is a very good thing to keep!!